sqlx-core/src/postgres/types/ltree.rs

```diff
@@ -0,0 +1,130 @@
+use crate::decode::Decode;
+use crate::encode::{Encode, IsNull};
+use crate::error::BoxDynError;
+use crate::postgres::{PgArgumentBuffer, PgTypeInfo, PgValueFormat, PgValueRef, Postgres};
+use crate::types::Type;
+use std::fmt::{self, Display, Formatter};
+use std::str::FromStr;
+
+/// Represents ltree specific errors
+#[derive(Debug, thiserror::Error)]
+#[non_exhaustive]
+pub enum Error {
+    /// LTree labels can only contain [A-Za-z0-9_]
+    #[error("ltree label cotains invalid characters")]
+    InvalidLtreeLabel,
+
+    /// LTree version not supported
+    #[error("ltree version not supported")]
+    InvalidLtreeVersion,
+}
+
+/// Represents an postgres ltree. Not that this is an EXTENSION!
+#[derive(Clone, Debug, Default, PartialEq)]
+pub struct PgLTree {
+    labels: Vec<String>,
+}
+
+impl PgLTree {
+    /// creates default/empty ltree
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// creates ltree from a [Vec<String>] without checking labels
+    pub fn new_unchecked(labels: Vec<String>) -> Self {
+        Self { labels }
+    }
+
+    /// creates ltree from an iterator with checking labels
+    pub fn from_iter(
+        labels: impl IntoIterator<Item = String>,
+    ) -> Result<Self, Error> {
+        let mut ltree = Self::default();
+        for label in labels.into_iter() {
+            ltree.push(label)?;
```

Comment:
For more flexibility, how about:

```suggestion
    pub fn from_iter<I, S>(labels: I) -> Result<Self, Error> 
    where
        S: Into<String>,
        I: IntoIterator<Item = S> 
    {
        let mut ltree = Self::default();
        for label in labels {
            ltree.push(label.into())?;
```

Reply:
(This would actually be pretty useful for testing with string literals.)